tests/std/tests/P0896R4_ranges_alg_partial_sort_copy/test.cpp

```diff
@@ -0,0 +1,171 @@
+
+            { // Validate iterator overload
+                for (const int i : sizes) {
+                    In range1{source};
+                    Out range2{span{output}.first(static_cast<size_t>(i))};
+                    const same_as<partial_sort_copy_result<iterator_t<In>, iterator_t<Out>>> auto result =
+                        partial_sort_copy(
+                            range1.begin(), range1.end(), range2.begin(), range2.end(), less{}, get_first, get_first);
+                    assert(result.in == range1.end());
+                    const auto n = min(i, int{size(source)});
+                    assert(result.out == range2.begin() + n);
+                    assert(equal(range2.begin(), range2.begin() + n, expected, expected + n));
+                }
+
+                // also with empty input
+                In range1{};
+                Out range2{output};
+                const same_as<partial_sort_copy_result<iterator_t<In>, iterator_t<Out>>> auto result =
+                    partial_sort_copy(
+                        range1.begin(), range1.end(), range2.begin(), range2.end(), less{}, get_first, get_first);
+                assert(result.in == range1.end());
+                assert(result.out == range2.begin());
+            }
+        }
+    }
+};
+
+#ifdef TEST_EVERYTHING
+int main() {
+    // No constexpr tests - these overrun the compilers' constexpr step limits quickly
+    test_in_random<instantiator1, const P, P>();
+    test_in_random<instantiator2, const P, P>();
+}
+#else // ^^^ test all range combinations // test only interesting range combos vvv
+constexpr void run_tests() {
+    using namespace test;
+    using test::iterator, test::range;
+    // The algorithm uses advance(i, s) in the input range, so it's minorly sensitive to that range's commonality and/or
+    // difference capability. We therefore test three kinds of source ranges:
+    using source_input = range<input, const P, Sized::no, CanDifference::no, Common::no, CanCompare::no, ProxyRef::yes>;
+    using source_forward =
+        range<fwd, const P, Sized::no, CanDifference::no, Common::yes, CanCompare::yes, ProxyRef::yes>;
+    using source_random =
+        range<random, const P, Sized::no, CanDifference::yes, Common::no, CanCompare::yes, ProxyRef::yes>;
+
+    // The result range must be random access - it's passed to various heap algorithm internals. Let's go ahead and use
+    // all 15 permutations of random access (or contiguous) ranges:
+    with_random_ranges<instantiator1, P>::call<source_input>();
+    with_random_ranges<instantiator1, P>::call<source_forward>();
+    with_random_ranges<instantiator1, P>::call<source_random>();
+
+    with_random_ranges<instantiator2, P>::call<source_input>();
+    with_random_ranges<instantiator2, P>::call<source_forward>();
+    with_random_ranges<instantiator2, P>::call<source_random>();
+}
+
+struct weird_pair : public pair<string, string> {

```

Comment:
We conventionally omit `public` for `struct` inheritance.
```suggestion
struct weird_pair : pair<string, string> {
```